### ISSUE TITLE
Code Review for Odometry and PID

### ISSUE THREAD

**Issue by jreyes81** [CSULATeleop/teleop#10]
Hey guys, I wanted to know if you could take a look at the odometry code and PID controller code I have for the robot. There are a few things i still need to calculate and a few other things I am iffy about. 

One of the things i was thinking of changing was the "cmd_vel" topic. Currently, the twist messages are being sent over the "cmd_vel" topic and the arduino picks it up. With the PID controller being implemented , I was thinking of having the "cmd_vel" topic going to the PID and having "cmd_vel_ard" as the topic being published from the PID node. This way the arduino will obtain info from it .

Let me know what you guys think. Thanks

**Comment by jay3ss** [comment]
Are we using a Raspberry Pi now? I notice that these are written in Python.

**Comment by jreyes81** [comment]
No. I just decided to use python is all

**Comment by jay3ss** [comment]
So, a PID isn't going to be running on the Arduino?

**Comment by jreyes81** [comment]
No. The PID will be handled by the laptop. The arduino will just count and publish the ticks from the encoders.

**Comment by jay3ss** [comment]
What is the PID operating on?

**Comment by jreyes81** [comment]
It compares the desired ("cmd_vel") velocity from the joystick input and the actual linear and angular velocity that is calculated from the odometry code based on the encoders. I also uploaded the updated arduino differential drive code labeled "differential_drive_encoder.ino"

**Comment by jreyes81** [comment]
Im not sure if thats what you mean

**Comment by jay3ss** [comment]
OK, that makes sense.

**Comment by jay3ss** [comment]
This is all in the master branch?

**Comment by jreyes81** [comment]
I uploaded it to the "GUI_Dev" branch in the src folder
